### ISSUE TITLE
0.4.0 breaks compatibility with test-unit 2.x

### ISSUE THREAD

**Issue by exterm** [alto/assert_json#10]
The update to 0.4.0 broke our test suite.

```
/home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/assert_json-0.4.0/lib/assert_json/assert_json.rb:12:in `size': wrong number of arguments (0 for 1) (ArgumentError)
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/testsuite.rb:82:in `block in size'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/testsuite.rb:82:in `each'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/testsuite.rb:82:in `size'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/testsuite.rb:87:in `empty?'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/testsuitecreator.rb:32:in `create'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/testcase.rb:139:in `suite'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/collector.rb:60:in `block in add_test_case'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/collector.rb:59:in `each'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/collector.rb:59:in `add_test_case'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/collector.rb:35:in `block in add_test_cases'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/collector.rb:34:in `each'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/collector.rb:34:in `add_test_cases'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/collector/descendant.rb:13:in `collect'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/autorunner.rb:84:in `block in <class:AutoRunner>'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/autorunner.rb:400:in `[]'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/autorunner.rb:400:in `run'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit/autorunner.rb:59:in `run'
	from /home/philip/.rvm/gems/ruby-2.2.2@avocado/gems/test-unit-2.5.5/lib/test/unit.rb:502:in `block (2 levels) in <top (required)>'
rake aborted!
```

The corresponding lines in test-unit are
```
      def size                                                                                                        
        total_size = @n_tests                                                                                         
        @tests.each { |test| total_size += test.size }                                                                
        total_size                                                                                                    
      end                                                                                                             
```

So it seems that the new size method in `assert_json 0.4.0` somehow overwrites the `test.size` used in test-unit.

**Comment by exterm** [comment]
Seems like this method (https://github.com/alto/assert_json/commit/98d1b360f2ea8706fd45e3364033f6b0a64c7bb3#diff-3e5ddb805e78926f54b08366213849f1R12) maybe should be called `json_size` or something like that.

**Comment by alto** [comment]
Will release a fix soon.

**Comment by alto** [comment]
I just released version `0.4.1`, including the patch provided by @thomasjachmann . We renamed the method from `size` to `has_length_of`.

**Comment by alto** [comment]
@exterm Thanks for pointing that out!
